### PR TITLE
Tests + Refactor

### PR DESCRIPTION
**Pull request by liorblob**
Fixed validation
Moved parser definition to classes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
...
+```
+To run on a specific environment, you can use:
+```
+tox -q -e py37
+```
+
+
+## License
+MIT - See [LICENSE](LICENSE) for more information.
+
```
Comment (Itay4) [inline README.md]: add contribution section

File: README.md [hunk 2]
```diff
@@ -1 +1,56 @@
...
+[![CircleCI](https://circleci.com/gh/demisto/demisto-sdk/tree/master.svg?style=svg)](https://circleci.com/gh/demisto/demisto-sdk/tree/master)
+
+# Demisto SDK 
+
+The Demisto SDK library can be used to manage your Demisto content with ease and efficiency.
+
+## Usage
+
+### Installation
+
```
Comment (Itay4) [inline README.md]: any prerequisites ?

File: README.md [hunk 3]
```diff
@@ -1 +1,56 @@
-# demisto-sdk
\ No newline at end of file
+[![CircleCI](https://circleci.com/gh/demisto/demisto-sdk/tree/master.svg?style=svg)](https://circleci.com/gh/demisto/demisto-sdk/tree/master)
+
+# Demisto SDK 
+
+The Demisto SDK library can be used to manage your Demisto content with ease and efficiency.
+
```
Comment (Itay4) [inline README.md]: write somewhere that only py3 is supported

File: demisto_sdk/core.py [hunk 4]
```diff
@@ -27,7 +31,34 @@ def __init__(self, configuration=Configuration()):
...
+        elif args.command == 'unify':
+            self.unify_package(args.indir, args.outdir)
+        elif args.command == 'validate':
+            if self.validate(is_backward_check=args.backward_comp, is_circle=args.circle,
+                             prev_ver=args.prev_ver, validate_conf_json=args.conf_json, use_git=args.use_git):
+                print_color('The files are valid', LOG_COLORS.GREEN)
+            else:
+                print_color('The files are invalid', LOG_COLORS.RED)
+        else:
+            print('Use demisto_sdk -h for help with the commands.')
```
Comment (Itay4) [inline demisto_sdk/core.py]:
```suggestion
            print('Unknown command, run `demisto_sdk -h` to see the available commands.')
```

File: demisto_sdk/yaml_tools/extractor.py [hunk 5]
```diff
@@ -163,3 +165,23 @@ def get_yml_type(self):
...
+                                 " Pass to -o option a directory in this case.")
+        parser.add_argument("-t", "--type",
+                            help="Yaml type. If not specified will try to determine type based upon path.",
+                            choices=[SCRIPT_PREFIX, INTEGRATION_PREFIX], default=None)
+        parser.add_argument("-d", "--demistomock", help="Add an import for demisto mock",
+                            choices=[True, False], type=str2bool, default=True)
+        parser.add_argument("-c", "--commonserver",
+                            help=("Add an import for CommonServerPython. "
+                                  " If not specified will import unless this is CommonServerPython"),
+                            choices=[True, False], type=str2bool, default=None)
```
Comment (Itay4) [inline demisto_sdk/yaml_tools/extractor.py]: why no default here?

File: README.md [hunk 6]
```diff
@@ -1 +1,56 @@
...
+### Installation
+
+`pip install demisto-sdk`
+
+### CLI
+You can use the SDK in the CLI as follows:
+`demisto_sdk <action> <args>`. For more information, run `demisto_sdk -h`.
+
+### In the code
+You can import the SDK core class in your code as follows:
```
Comment (Itay4) [inline README.md]: mention its in python code

File: README.md [hunk 7]
```diff
@@ -1 +1,56 @@
...
+
+## Usage
+
+### Installation
+
+`pip install demisto-sdk`
+
+### CLI
+You can use the SDK in the CLI as follows:
+`demisto_sdk <action> <args>`. For more information, run `demisto_sdk -h`.
```
Comment (Itay4) [inline README.md]: lets add some usage examples

File: README.md (hunk 2 above)
Comment (liorblob) [reply README.md]: yet to be determined... except unify/extract, it's only working on our content repo for now

File: demisto_sdk/core.py (hunk 4 above)
Comment (liorblob) [reply demisto_sdk/core.py]: this message is displayed also when someone types just `demisto_sdk`

**Pull request merged**